DemoApp/Sources/Extensions/CallViewModel+Snapshot.swift

```diff
@@ -20,6 +28,43 @@ extension CallViewModel {
             }
         }
     }
+
+    private func resize(
+        image: UIImage,
+        to targetSize: CGSize
+    ) -> UIImage? {
+        guard
+            image.size.width > targetSize.width || image.size.height > targetSize.height
+        else {
+            return image
+        }
+
+        let widthRatio = targetSize.width / image.size.width
+        let heightRatio = targetSize.height / image.size.height
+
+        // Determine the scale factor that preserves aspect ratio
+        let scaleFactor = min(widthRatio, heightRatio)
+
+        let scaledWidth = image.size.width * scaleFactor
+        let scaledHeight = image.size.height * scaleFactor
+        let targetRect = CGRect(
+            x: (
+                targetSize.width - scaledWidth
+            ) / 2,
+            y: (targetSize.height - scaledHeight) / 2,
+            width: scaledWidth,
+            height: scaledHeight
+        )
+
+        // Create a new image context
+        UIGraphicsBeginImageContextWithOptions(targetSize, false, 0)
+        image.draw(in: targetRect)
+
+        let newImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
```

Comment:
Since we turn this into a JPEG in the next step then I was wondering if we could benefit from: [UIGraphicsImageRenderer.jpegData(withCompressionQuality:actions:)](https://developer.apple.com/documentation/uikit/uigraphicsimagerenderer/1649234-jpegdata)
But in that case we probably should have the targetRect logic separated as well for making it easier to use it with jpeg creation method and the image creation method on UIGraphicsImageRenderer.

Reply:
Later on we are using it (via UIImage) but we don't have "hard" requirements here. We are mostly doing it demonstrate sending data through our WebRTC stack (which poses a 100Kb size limitation and thus we the reduction in quality).